ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/PendingTransaction.java

```diff
@@ -38,7 +38,9 @@ public abstract class PendingTransaction
   static final int PAYLOAD_BASE_MEMORY_SIZE = 32;
   static final int ACCESS_LIST_STORAGE_KEY_MEMORY_SIZE = 32;
   static final int ACCESS_LIST_ENTRY_BASE_MEMORY_SIZE = 248;
-  static final int OPTIONAL_ACCESS_LIST_MEMORY_SIZE = 24;
+  static final int OPTIONAL_ACCESS_LIST_MEMORY_SIZE = 40;
+  static final int OPTIONAL_CODE_DELEGATION_LIST_MEMORY_SIZE = 40;
+  static final int CODE_DELEGATION_ENTRY_MEMORY_SIZE = 432;
```

Comment:
I have reorganized the code and added documentation on how to calculate these numbers and how to use the test to verify if the sizes needs to be updated. 
The result of JOL can be always produced just running the specific unit test, and I am afraid that adding the output to the description can be easily get stale, so I prefer to avoid that, and improve the readability of the code if needed